addons/stock_picking_batch/models/stock_picking_batch.py

```diff
@@ -227,11 +227,12 @@ def action_print(self):
         return self.env.ref('stock_picking_batch.action_report_picking_batch').report_action(self)
 
     def action_done(self):
+        precision_digits = self.env['decimal.precision'].precision_get('Product Unit of Measure')
         def has_no_quantity(picking):
```

Comment:
Both should have an extra parameter for the precision digits, then fetch the precision after the `ensure_one`/`_check_company` are done